doc/HLD_Meta_Data_Back_End.rst

```diff
@@ -0,0 +1,110 @@
+
+*********************
+Introduction
+*********************
+
+Meta-data back-end (BE) is a module presenting an interface for a transactional local meta-data storage. BE users manipulate and access meta-data structures in memory. BE maps this memory to persistent storage. User groups meta-data updates in transactions. BE guarantees that transactions are atomic in the face of process failures. BE provides support for a few frequently used data-structures: double linked list, B-tree and extmap.
+
+*********************
+Dependencies
+*********************
+
+- a storage object (stob) is a container for unstructured data, accessible through m0_stob interface. BE uses stobs to store meta-data on persistent store. BE accesses persistent store only through m0_stob interface and assumes that every completed stob write survives any node failure. It is up to a stob implementation to guarantee this;
+
+- a segment is a stob mapped to an extent in process address space. Each address in the extent uniquely corresponds to the offset in the stob and vice versa. Stob is divided into blocks of fixed size. Memory extent is divided into pages of fixed size. Page size is a multiple of block size (it follows that stob size is a multiple of page size). At a given moment in time, some pages are up-to-date (their contents is the same as of the corresponding stob blocks) and some are dirty (their contents was modified relative to the stob blocks). In the initial implementation all pages are up-to-date, when the segment is opened. In the later versions, pages will be loaded dynamically on demand. The memory extent to which a segment is mapped is called segment memory; 
+
+- a region is an extent within segment memory. A (meta-data) update is a modification of some region;
+
+- a transaction is a collection of updates. User adds an update to a transaction by capturing the update's region. User explicitly closes a transaction. BE guarantees that a closed transaction is atomic with respect to process crashes that happen after transaction close call returns. That is, after such a crash, either all or none of transaction updates will be present in the segment memory when the segment is opened next time. If a process crashes before a transaction closes, BE guarantees that none of transaction updates will be present in the segment memory;
+
+- a credit is a measure of a group of updates. A credit is a pair (nr, size), where nr is the number of updates and size is total size in bytes of modified regions.
+
+*********************
+Requirements
+*********************
+
+- R.C2.MDSTORE.NUMA: allocator respects NUMA topology 
```

Comment:
Make the font BOLD:
```
*R.C2.MDSTORE.NUMA*
```

Reply:
And it's better to rename "C2" to "M0"

Reply:
Incorporated.

Reply:
I mean, to make all the requirements in this section BOLD.

Reply:
![Requirements](https://user-images.githubusercontent.com/68016036/100324050-a9a6cb80-2fec-11eb-85a9-311e20c745a0.PNG)

Reply:
The requirements headline is bolded.

Reply:
Incorporated.